jplag.endToEndTesting/src/test/java/jplag/endToEndTesting/javaTestCases.java

```diff
@@ -0,0 +1,242 @@
+
+			List<String> tempList = new ArrayList<String>();
+			tempList.add(testSuiteHelper.getFolderPath());
+
+			JPlagOptions options = new JPlagOptions(tempList, new ArrayList<String>(), LanguageOption.JAVA);
+
+			JPlag jplag = new JPlag(options);
+			JPlagResult result = jplag.run();
+			// after close the created directories are deleted
+			testSuiteHelper.close();
+
+			ResultComparison resultComparison = new ResultComparison(new Object() {
+			}.getClass().getEnclosingMethod().getName());
+			// for the fine-granular testing strategy, the result object contains only one
+			// comparison.
+			for (JPlagComparison jPlagComparison : result.getAllComparisons()) {
+				System.out.println("Comparison of the stored values and the current equality values");
+				assertEquals(resultComparison.getSimilarity(), jPlagComparison.similarity());
+			}
+
+		} catch (Exception e) {
+			fail(e.getMessage());
+		}
+	}
+
+	/**
+	 * Insertion of unnecessary or changed code lines (token generation)
+	 */
+	@Test
+	void tokenGenerationLevelTest_one() {
+		String[] testClassNames = new String[] { "SortAlgo.java", "SortAlgo3.java" };
+
+		try {
+			var testSuiteHelper = new JPlagTestSuiteHelper(testClassNames);
+
+			List<String> tempList = new ArrayList<String>();
+			tempList.add(testSuiteHelper.getFolderPath());
+
+			JPlagOptions options = new JPlagOptions(tempList, new ArrayList<String>(), LanguageOption.JAVA);
+
+			JPlag jplag = new JPlag(options);
+			JPlagResult result = jplag.run();
+			// after close the created directories are deleted
+			testSuiteHelper.close();
+
+			ResultComparison resultComparison = new ResultComparison(new Object() {
+			}.getClass().getEnclosingMethod().getName());
+			// for the fine-granular testing strategy, the result object contains only one
+			// comparison.
+			for (JPlagComparison jPlagComparison : result.getAllComparisons()) {
+				System.out.println("Comparison of the stored values and the current equality values");
+				assertEquals(resultComparison.getSimilarity(), jPlagComparison.similarity());
+			}
+
+		} catch (Exception e) {
+			fail(e.getMessage());
+		}
+	}
+
+	/**
+	 * Changing the program flow (token generation) (statments and functions must be
+	 * independent from each other)
+	 */
+	@Test
+	void tokenGenerationLevelTest_two() {
+		String[] testClassNames = new String[] { "SortAlgo.java", "SortAlgo4.java" };
+
+		try {
+			var testSuiteHelper = new JPlagTestSuiteHelper(testClassNames);
+
+			List<String> tempList = new ArrayList<String>();
+			tempList.add(testSuiteHelper.getFolderPath());
+
+			JPlagOptions options = new JPlagOptions(tempList, new ArrayList<String>(), LanguageOption.JAVA);
+
+			JPlag jplag = new JPlag(options);
+			JPlagResult result = jplag.run();
+			// after close the created directories are deleted
+			testSuiteHelper.close();
+
+			ResultComparison resultComparison = new ResultComparison(new Object() {
+			}.getClass().getEnclosingMethod().getName());
```

Comment:
This code is duplicated in every test case.

Reply:
Yes that's right. I plan to make the helper class static and thus remove the duplicate code in the test cases

Reply:
Test cases were changed to remove the duplicate code lines:

``` java
/**
	 * Inserting comments or empty lines (normalization level)
	 * 
	 * @throws Exception
	 */
	@Test
	void normalizationLevelTest_one() throws Exception {

		String[] testClassNames = new String[] { "SortAlgo.java", "SortAlgo1.java" };

		TestCaseModel testCaseModel = jplagTestSuiteHelper.createNewTestCase(testClassNames);

		JPlagResult jplagResult = new JPlag(testCaseModel.getJPlagOptionsFromCurrentModel()).run();

		assertTrue(testCaseModel.compaireModelProperties(jplagResult), "does not match");
	}
```

Reply:
Looks good!

Reply:
Done. 
What I wanted to add is that the function names currently act as identifiers in the *.json file. Changes to a hash would be possible, but the readability of the *.json file is currently better than with other identifiers.

Reply:
Jackson JSON allowes the definition of the names to be used in JSON files.